web/models.py

```
@@ -1752,6 +1752,64 @@ def display_name(self):
+        # Validate that location is provided for in-person meetups
+        if self.event_type == "in_person" and not self.location:
+            raise ValidationError("In-person meetups require a location")
+
+    def save(self, *args, **kwargs) -> None:
+        self.full_clean()  # Call full_clean to enforce validation
+        if not self.slug:
+            self.slug = slugify(self.title)
+            if Meetup.objects.filter(slug=self.slug).exists():
+                suffix = uuid.uuid4().hex[:6]
+                self.slug = f"{self.slug}-{suffix}"
+        super().save(*args, **kwargs)
```

Ruff (0.8.2)

Order of model's inner classes, methods, and fields does not follow the Django Style Guide: `save` method should come before custom method

(DJ012)